numba_dpex/core/types/usm_ndarray_type.py

```diff
@@ -31,46 +31,53 @@ def __init__(
         aligned=True,
         addrspace=address_space.GLOBAL,
     ):
+        # Creating SyclDevice from filter_string is expensive. So, USMNdArray should be able to
+        # accept and SyclDevice from usm_ndarray as device parameter
+        if not isinstance(device, (str, dpctl.SyclDevice)):
+            raise TypeError(
+                "The device keyword arg should be a str object specifying "
+                "a SYCL filter selector"
+            )
+
+        if not isinstance(queue, dpctl.SyclQueue) and queue is not None:
```

Comment:
```suggestion
        if queue and not isinstance(queue, dpctl.SyclQueue):
```

Reply:
I personally prefer more explicit `if queue is not None` than implicit `if queue`